backend/src/nodes/nodes/image_utility/convolve.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import annotations
+
+import numpy as np
+
+from . import category as ImageUtilityCategory
+from ...node_base import NodeBase
+from ...node_factory import NodeFactory
+from ...properties.inputs import ImageInput, NumberInput, TextInput
+from ...properties.outputs import ImageOutput
+from ...properties import expression
+
+
+@NodeFactory.register("chainner:image:image_convolve")
+class ImageConvolveNode(NodeBase):
+    def __init__(self):
+        super().__init__()
+        self.description = "Convolves input image with input kernel (kernel values separated by commas)"
+        self.inputs = [
+            ImageInput(),
+            TextInput("Kernel String"),
+            NumberInput("Kernel Width/Height", minimum=0, default=3),
+            NumberInput("Padding", minimum=0, default=0),
+            NumberInput("Strides", minimum=0, default=1),
+        ]
+        self.outputs = [
+            ImageOutput(image_type=expression.Image(size_as="Input0"), channels=1)
+        ]
+        self.category = ImageUtilityCategory
+        self.name = "Convolve"
+        self.icon = "MdAutoFixHigh"
+        self.sub = "Miscellaneous"
+
+    def run(
+        self,
+        img: np.ndarray,
+        kernel_in: str,
+        kernel_dim: int,
+        padding: int,
+        strides: int,
+    ) -> np.ndarray:
+
+        kernel = np.array([float(d) for d in kernel_in.split(",")])
+        kernel = kernel.reshape(kernel_dim, kernel_dim)
+
+        # Grayscale image if it is not already
+        if len(img.shape) != 2:
+            img = img[:, :, 0]
```

Comment:
I don't think that this is a good idea. Chainner stores color images as BGR internally, so you just get the B channel here.

Maybe we could just require grayscale images using [the `channels` parameter](https://github.com/chaiNNer-org/chaiNNer/blob/3fae92c05ffc94be5e9b84c0abcffa084bd6cff6/backend/src/nodes/properties/inputs/numpy_inputs.py#L24)? We could also use `as_target_channels(img, 1, narrowing=True)` to convert the image to grayscale in a way that is consistent with the rest of chainner.
Not sure what option is best here.

Reply:
Oh, that's a great point, I forgot about the other nodes as I was first testing the convolve in a Python notebook. I think it'll be best to require grayscale images in this case, I'll implement it now.